internal/proxy/proxy.go

```diff
@@ -181,6 +189,19 @@ type Config struct {
 	RunConnectionTest bool
 }
 
+// dialOptions interprets appropriate dial options for a particular instance
+// configuration
+func dialOptions(c Config, i InstanceConnConfig) []alloydbconn.DialOption {
+	var opts []alloydbconn.DialOption
+
+	// If public IP is enabled at the instance level, or public IP is enabled
+	// globally, add the option.
+	if i.PublicIP || c.PublicIP {
+		opts = append(opts, alloydbconn.WithPublicIP())
+	}
```

Comment:
This logic may need to be updated, depends on intended behavior. If someone has globally set `publicIP` to True  but query params it with false do we want the query param to override?

Reply:
Yes we do want the param to override in that case. Think connecting to three instances, all but one is public ip. You'd set the global flag and disable the flag with a query param for the private instance.

Reply:
Should be able to override now, added test to make sure